### DISCUSSION TITLE
Shadow jitter problem in large scenes

### DISCUSSION THREAD

**Discussion by wddl** [google/filament#7188] [Q&A]
The 3D model is placed in a large world where the shadows shake at a Specific camera height. And this phenomenon does not occur in a small scene. How should this problem be solved?Thanks

**Comment by bejado** [comment]
There's been some work on shadows recently, make sure you're using the latest `main` branch or wait for a more recent build of Filament to release. Can you post a video of the phenomenon? There are a lot of parameters in [`ShadowOptions`](https://github.com/google/filament/blob/cb9bfaeb83cb69f2643538404a81832b1458435d/filament/include/filament/LightManager.h#L192) that you can try adjusting. In particular, `stable` might be useful.

**Reply by wddl** [reply]
thanks! stable is useful, but it has a limited effect, and a slightly larger scene is not. I put the model on the scale of the earth. Recorded a video and sent it out. There was a serious problem at a specific camera height.

https://github.com/google/filament/assets/26518625/e8542e4f-2ba4-43b9-9c6d-5ed0797554fa

**Reply by wddl** [reply]
I haven't tested the latest version yet

**Reply by bejado** [reply]
@pixelflinger Might have some ideas for you to try. But definitely first get a recent build.

**Reply by wddl** [reply]
Thanks!   I'll test the latest version

**Reply by wddl** [reply]
@bejado I have tested version 1.43.1, but the problem still exists. It's strange that the shadow effect is fine when the camera is at a far distance, not in the middle, and in close range the effect is fine.
![image](https://github.com/google/filament/assets/26518625/73efe375-7564-4b77-901e-30a28193a5b5)
![image](https://github.com/google/filament/assets/26518625/51306ac4-dca1-438b-bba6-44eb65068bfc)
![image](https://github.com/google/filament/assets/26518625/d576ce52-ff05-423d-bea9-51a207eede35)
